### DISCUSSION TITLE
Unable to create questionset in VDN

### DISCUSSION THREAD

**Discussion by yashveertrigyn** [Sunbird-coKreat/Community#93] [Q&A] [answered]
I am getting issue when adding **Exam Question Sets**, getting validation issue "board range data is empty from the given framework.". in API **/action/questionset/v1/create**

I am logged in as hr_k-12_2 tenant user. I am getting all the framework values for user but trying to add question sets then getting validation error for framework validation.

We checked learner/user/v5/read/{Id} API response then checked **/action/questionset/v1/create** 

API with request  **/action/questionset/v1/create** 
------------------------------------------------------
**{"request":{"questionset":{"name":"Untitled","code":"fdb0b4fb-3f1c-5d3f-2efc-6215ee0f98c2","mimeType":"application/vnd.sunbird.questionset","createdBy":"bab03a78-420b-49af-a6f2-8f3e1d068325","primaryCategory":"Exam Question Set","creator":"SABIR AHMED","author":"SABIR AHMED","programId":"09980080-4e54-11ee-99c0-5b5c86bd6067","channel":"01257280950159769620","framework":"hr_k-12_2","board":"State (Haryana)","medium":[],"gradeLevel":[],"subject":[],"boardIds":[],"mediumIds":[],"gradeLevelIds":[],"subjectIds":[],"topic":[],"learningOutcome":[],"learningOutcomeIds":[]}}}**

And Response we are getting.
-------------------------------

**{
    "id": "api.questionset.create",
    "ver": "3.0",
    "ts": "2023-09-08T14:29:01ZZ",
    "params": {
        "resmsgid": "680edfc5-255c-40b1-9f68-1c3367d9e656",
        "msgid": null,
        "err": "CLIENT_ERROR",
        "status": "failed",
        "errmsg": "Validation Errors."
    },
    "responseCode": "CLIENT_ERROR",
    "result": {
        "messages": [
            "board range data is empty from the given framework."
        ]
    }
}**

The above error we are getting from **Redis** when it is trying to get Board data from **Redis** Cache. Can anyone help in this?
![Screenshot 2023-09-08 200440](https://github.com/Sunbird-coKreat/Community/assets/141142203/2d4a052e-6e5d-40d4-9a1e-f4899bca4da6)

**Comment by Aparna-karri591** [comment]
[yashveertrigyn](https://github.com/yashveertrigyn) Can you Please let us know on which Environment you are trying this? Dev or Staging?

**Comment by yashveertrigyn** [comment]
@Aparna-karri591 I am trying in production. We are facing this issue in production.

**Reply by vinukumar-vs** [reply]
@yashveertrigyn What is your analysis on this? 
Message is clearly saying "board range data is empty from the given framework." 

Did you verified the framework & question-set schema to verify why the board is invalid?
@vaivk369 @Aparna-karri591

**Reply by yashveertrigyn** [reply]
Hi @vinukumar-vs 

We have checked the following question-set schema in the code base and blob storage as well and compared it with framework that we are getting for the logged in user, the board values are present for the logged in user but the system still gives an validation error.

 
The result of the user read call gives the following framework output:

"framework": {
        "board": [
          "State (Haryana)"
        ],
        "gradeLevel": [
          "Class 3",
          "Class 4"
        ],
        "id": [
          "hr_k-12_2"
        ],
        "medium": [
          "English",
          "Hindi"
        ],
        "subject": []
      },

And the question-set schema also has the board category as shown below:

Schema for question set- 
[question-set-schema.txt](https://github.com/Sunbird-coKreat/Community/files/12572950/question-set-schema.txt)

**Reply by vinukumar-vs** [reply, marked as answer]
[yashveertrigyn](https://github.com/yashveertrigyn) as we discussed in the call, please disable the schema validation configuration for assessment-service.

refer the below configuration
https://github.com/project-sunbird/sunbird-devops/blob/b428c1a3cf930325a7bd617bad358b50ff5894d9/ansible/roles/stack-sunbird/templates/assessment-service_application.conf#L409

Hope this solved the issue. Hence I have marked as Answered directly. If not please feel free to add your comments to reopen the issue.
@vaivk369 @Aparna-karri591 @santhosh-tg
